test/box/update.test.lua

```diff
@@ -629,7 +629,7 @@ t:upsert(ops)
 -- gh-8216: attempt to make an update inside a just created array field could
 -- crash.
 --
-t:update({{'=', '[3]', {a = 33}}, {'+', '[3].a', 1}})
+t:update({{'=', '[2]', {a = 33}}, {'+', '[2].a', 1}})
```

Comment:
The thing is if I keep this test unchanged then after the patch I'll get:
```
---
- error: 'Tuple field 3 (g) type does not match one required by operation: expected
    array, got map'
...
```
It is not from update itself. Update will produce `{a=34}` in the second field of the tuple. But tuple format states second field should be an array. So test become a bit strange.

Reply:
It might be that this test in its new form won't cover the crash fix done in the same commit where the test was introduced. That is why I would propose to keep it as is even if it fails weirdly. It just shouldn't crash. It is not expected to succeed AFAIR.

Reply:
Fixed.